### DISCUSSION TITLE
Handling encryption at rest?

### DISCUSSION THREAD

**Discussion by mjyoung** [prisma/prisma#6616] [Q&A] [answered]
With prisma (and postgres), how do you all handle columns or tables that need to be encrypted at rest (if HIPAA/etc. compliance is important)? Examples include PHI or PII data such as SSN, DOB, Address, etc?

Any strategies would be much appreciated! Wondering how queries are handled as well if things are encrypted at rest.

Or should the encryption just handle on the nodejs side and inserted into the database pre-encrypted

**Comment by VinayaSathyanarayana** [comment]
Prisma Middleware is an option https://www.prisma.io/docs/concepts/components/prisma-client/middleware

**Reply by ryands17** [reply, marked as answer]
No middleware is not an option. For encryption at rest the server cannot know what data you're sending so you always need to encrypt it client side. So Prisma cannot be helpful in this case unfortunately.

**Comment by VinayaSathyanarayana** [comment]
Is Prisma Middleware - a client side or on server side?

**Reply by ryands17** [reply]
That would be server side.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
